### PR TITLE
UX suggested edits

### PR DESCRIPTION
**Pull request by bensaufley**
My suggestions based on current implicit (and discussed) styles and best practices. As you can see, there are a few things oppose Thoughtbot's styles, but most of it is more extensions than revisions.

I had a little trouble differentiating between Styles and Best Practices, so … let me know if you think I should move some stuff.

I will leave this open for a week to gather comments and revisions.

### CODE REVIEW

**Comment by fishermanswharff** [comment]
Best Practice: use CSS3 animations instead of jQuery animations

**Comment by fishermanswharff** [comment]
Best Practice: use `translate3d(x,y,z)` to offload animations to GPU on hardware

**Comment by bensaufley** [comment]
I'm in favor of CSS Animations where possible, except it's a little trickier because of IE8, right? If the animation has real use (and isn't just an aesthetic bonus), it needs to be run in jQuery in browsers that don't support CSS Animations. I think this falls under "graceful degradation/progressive enhancement" generally, which I did add.

**Comment by fishermanswharff** [comment]
yes, trickier with IE8, maybe jQuery animations for ie8, CSS3 animations for everything else?

**Comment by kylepfeeley** [comment]
I would say IE8 just doesn't get animations at all unless it's vital to the UI.

**Comment by fishermanswharff** [comment]
+1 on that @kylepfeeley

**Comment by bensaufley** [comment]
The other thing is slideUp and slideDown don't really have a clean CSS3 analog, and we use those all over the place. It's possible to animate `max-height`, but that's not really clean and unless you have a fairly predictable height range, it's not really practical.




I'm not looking to reject ideas to the style guide, but I think there are so many ifs about this that I'm not sure there's really a rule. Have to use best judgment.




@kylepfeeley, yeah, that's what I was trying to say.


--

Ben Saufley

Technical Lead, UX

EverQuote, Inc

**Comment by bensaufley** [comment]
Added HAML style guide, which got left out of initial commit despite being there. Made updates based on @fishermanswharff's suggestions

**Comment by fishermanswharff** [comment]
how do we feel about single-line declarations like so: 
```
h4 { text-align: center }

img {
  display: block;
  …
}
```

**Comment by bensaufley** [comment]
We have used them in the past but I think it's an unnecessary complication and easy to avoid. That said, I'm a big fan of trailing conditionals and ternary operators, which have some of the same utility, so I could be persuaded. I just figured it wasn't worth the extra rules. Just keep everything consistent.

**Comment by akainic** [comment]
I don't like single-line declarations. If we have a convention of using separate lines for most cases we should do that everywhere, just makes it easier to read if it's consistent.

**Comment by bensaufley** [comment]
Yeah. That was my thought. I think there's an argument for something like below, but I don't really think it's worth the trouble. Easier to just say "don't". Also works better with SCSS-Lint than allowing in *some* cases but not others. Can just say "don't allow it."

```scss
.thing { @extend .other-thing; }
// or
.thing { @include thing-mixin(); }
```

**Comment by bensaufley** [comment]
Leaving open till Friday per @akainic's request. Will merge on Friday, barring objections.

File: best-practices/README.md [hunk 1]
```diff
@@ -197,25 +206,29 @@ HTML
 * Don't use a reset button for forms.
 * Prefer cancel links to cancel buttons.
 * Use `<button>` tags over `<a>` tags for actions.
+* Place JavaScript at the foot of the document when possible.
```
Comment (akainic) [inline best-practices/README.md]: Maybe specify in the `:deferred_content` content block?

File: style/coffeescript/README.md [hunk 2]
```diff
@@ -11,8 +10,8 @@ CoffeeScript
 * Prefer `||` and `&&` to `or` and `and`
 * Prefer `!` over `not`
 * Prefer `@` over `this` for referencing instance properties.
-* Prefer double quotes.
-* Use hyphen-separated filenames, such as `coffee-script.coffee`.
+* Use single quotes for non-interpolated strings.
+* Use hyphen-separated filenames, such as `coffee-script.js.coffee`.
 * Use `PascalCase` for classes, `lowerCamelCase` for variables and functions,
```
Comment (akainic) [inline style/coffeescript/README.md]: Wait, don't we use `train-case` or whatever it's called?

File: style/html/README.md [hunk 3]
```diff
@@ -1,4 +1,23 @@
 HTML
 ====
```
Comment (akainic) [inline style/html/README.md]: I know we don't currently have a naming convention for classes, but is that something we should think about?

File: style/sass/README.md [hunk 4]
```diff
@@ -33,24 +42,32 @@
...
+* Avoid redundant classnames: `article.main` not `article.main-article` and `.button.green` not `.button.green-button`.
+* Avoid nesting more than 3 selectors deep.
+* Don't nest more than 5 selectors deep.
 * Use HTML tags on vague classes that need a qualifier like `header.application` not `.main`.
 * Avoid using the HTML tag in the class name: `section.news` not `section.news-section`.
 * Avoid using HTML tags on classes for generic markup `<div>`, `<span>`: `.widgets` not `div.widgets`.
-* Avoid using HTML tags on classes with specific class names like `.featured-articles`.
-* Avoid using comma delimited selectors.
-* Avoid nesting within a media query.
+* Avoid using HTML tags on classes with specific class names: `.featured-articles` not `section.featured-articles`.
```
Comment (akainic) [inline style/sass/README.md]: Should we specify the case we using for naming classes?

File: best-practices/README.md (hunk 1 above)
Comment (bensaufley) [inline best-practices/README.md]: `:deferred_content` isn't always available, but I can probably add a note for Lynx.

File: style/html/README.md (hunk 3 above)
Comment (bensaufley) [inline style/html/README.md]: We do have this:

> Use hyphens when naming mixins, extends, classes, functions & variables: `span-columns` not `span_columns` or `spanColumns`.

In [SASS]. Are you looking for something more specific?

Comment (akainic) [inline style/html/README.md]: Well everywhere else when we specify naming conventions we identify the case we use so I just thought we should do the same thing here and describe it as a particular case.

Comment (bensaufley) [inline style/html/README.md]: So just adding "Use dash-case" is what you're looking for? I can see that that's more clear.

**Pull request merged**